pkg/kube/client.go

```diff
@@ -369,13 +370,55 @@ func batchPerform(infos ResourceList, fn func(*resource.Info) error, errs chan<-
 			kind = currentKind
 		}
 		wg.Add(1)
-		go func(i *resource.Info) {
-			errs <- fn(i)
+		go func(info *resource.Info) {
+			// try 3 times total
+			tries := 3
+			rand.Seed(time.Now().UTC().UnixNano())
+
+			fnErr := fn(info)
+			for counter := 1; counter < tries; counter++ {
+				// break out if this is not a Resource Quota Conflict of if the resource itself
+				// is a resourcequota
+				if fnErr == nil ||
+					kind == "resourcequotas" ||
+					!isResourceQuotaConflict(fnErr) {
+					break
+				}
+
+				// Increasing wait: 10-19ms, 40-59ms
+				waitTime := time.Duration(counter*counter*10+rand.Intn(counter*10)) * time.Millisecond
+				time.Sleep(waitTime)
+
+				// rerun fn(info)
+				fnErr = fn(info)
+			}
+			errs <- fnErr
 			wg.Done()
 		}(info)
 	}
 }
 
+// Helper function which checks if error e is a ResourceQuota conflict
+//
+// A ResourceQuota conflict is hit when the following 3 conditions are met:
+// a) Is this a StatusError?
+// b) Is this a Conflict?
+// c) Is status.Details.Kind == "resourcequotas"
+func isResourceQuotaConflict(e error) bool {
+	// Is this a StatusError?
```

Comment:
```golang
var (
  err *apierrors.StatusError
  ok = true
)
err, ok = e.(*apierrors.StatusError)
if !ok {
  return false
}
ok = apierrors.IsConflict(err)
if !ok {
  return false
}
status := err.Status()
return status.Details.Kind == "resourcequotas"
```
I think this is too much nested.
You can change like this

Reply:
I adjusted this